### PR TITLE
User project associations #50

### PR DESCRIPTION
**Pull request by pospieszynski**
I have set relations between user and project + added some specs to check if it works. Review please;]

### CODE REVIEW

File: app/models/project.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
 class Project < ApplicationRecord
   validates :title, :date, :description, presence: true
+  belongs_to :user
+  has_many :user_projects
+  has_many :contributors, :through => :user_projects, source: :user
```
Comment (neonowy) [inline app/models/project.rb]: I think you can write `through: :user_projects` instead.

File: db/migrate/20160720194304_add_user_id_to_projects.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserIdToProjects < ActiveRecord::Migration[5.0]
+  def change
+    add_column :projects, :user_id, :integer
```
Comment (filip373) [inline db/migrate/20160720194304_add_user_id_to_projects.rb]: I think you missed foreign key reference here. It only adds integer column `user_id` without related database constraint. Take a look http://stackoverflow.com/a/31882059

File: db/migrate/20160720224920_add_unique_contraint_to_user_project.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUniqueContraintToUserProject < ActiveRecord::Migration[5.0]
+  def change
+    add_index :user_projects, [:user_id, :project_id], unique: true
```
Comment (filip373) [inline db/migrate/20160720224920_add_unique_contraint_to_user_project.rb]: I'd merge this migration with `CreateUserProjects` as it's part of this table.

File: spec/models/project_spec.rb [hunk 4]
```diff
@@ -1,17 +1,74 @@
...
   end
 
   context "with valid params" do
     it "saves projects when all params are given" do
-      project = Project.new(title: "nothing", date: "20-07-2016", description: "about nothing")
       expect(project).to be_valid
     end
   end
+
+  context "Properly setup association - project belongs to user" do
```
Comment (filip373) [inline spec/models/project_spec.rb]: Use `with` and `when` in contexts, without capital letters.

**Comment by crappylime** [comment]
:+1:

**Comment by tprochniewicz** [comment]
@gotar

**Comment by gotar** [comment]
squash and you can merge

File: spec/models/user_project_spec.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  context "with proper parameters" do
+    it "Adds new contributor to project" do
+      user_project = UserProject.create(user: employee, project: project)
+      expect(user_project).to be_valid
+    end
+
+    it "doesnt add duplicated record" do
+      expect {
+        2.times { UserProject.create(user: employee, project: project) }
+      }.to raise_error
```
Comment (tprochniewicz) [inline spec/models/user_project_spec.rb]: you can check if it is ActiveRecord::RecordNotUnique error

**Pull request merged**